### DISCUSSION TITLE
How does projoint works with constraints, especially those that make one attribute to always differ from the attribute on the other side?

### DISCUSSION THREAD

**Discussion by brueckmann** [yhoriuchi/projoint#37] [Q&A] [answered]
I wanted to utilize the projoint out of sample prediction of IRR, i.e. no repeated or swapped task.

What is puzzling to me: When using conjoint data with 4 attributes, whereby 3 attributes are randomized completely random and one attribute is randomized in a constrained way to ensure that both profiles are always differing in this variable, the estimated IRR are much larger when the additional constrained variable is added to the model. I wonder why this is, as this attribute is not likely to add precision but rather by design prevents that any conjoint with both sides identical are displayed. 

I am currently trying to understand whether this is a feature or am I just really unlucky with my data?

To see where this problem is coming from, I tried to regroup the variables that are very close to each other in a way to sometimes have "as if" equal attribute levels on both side of the constraint attribute, which was more an exercise to see what would happen. When performing the regrouping and making the regrouped and therefore "less" constrained attribute one of the attributes in a new projoint  I got just the exact same IRRs as before where there were by design far less opportunities for equality. I guess this must be due to the weights, am I right? 

In case you can share any insights on this, that would be highly appreciated! And I just hope that this discussion with other users and creators might benefit the weider community of conjoint I mean projoint users :)

**Comment by yhoriuchi** [comment, marked as answer]
Hi. I used `exampleData1` to check whether IRR is different depending on whether one of the attribute is "tied" or not. 
```
library(tidyverse)
library(projoint)

outcomes <- paste0("choice", seq(from = 1, to = 8, by = 1))
outcomes <- c(outcomes, "choice1_repeated_flipped")
out1 <- reshape_projoint(exampleData1, outcomes)

check_IRR <- function(.att_number){
  
  var <- str_c("att", .att_number)
  var_quo <- rlang::sym(var)
  
  reshaped_data <- out1@data %>% 
    group_by(id, task) %>% 
    mutate(tie = case_when({{var_quo}} == lag({{var_quo}}) | {{var_quo}} == lead({{var_quo}}) ~ 1,
                           {{var_quo}} != lag({{var_quo}}) | {{var_quo}} != lead({{var_quo}}) ~ 0)) %>% 
    ungroup() %>% 
    filter(!is.na(agree))
  
  reshaped_data %>% 
    group_by(tie) %>% 
    summarise(irr = mean(agree)) %>% 
    ungroup() %>% 
    mutate(att = .att_number) %>% 
    relocate(att)
  
}

map_dfr(1:6, check_IRR)
```
The output is the following:
```
# A tibble: 12 × 3
     att   tie   irr
   <int> <dbl> <dbl>
 1     1     0 0.698
 2     1     1 0.746
 3     2     0 0.710
 4     2     1 0.726
 5     3     0 0.715
 6     3     1 0.714
 7     4     0 0.744
 8     4     1 0.687
 9     5     0 0.725
10     5     1 0.686
11     6     0 0.716
12     6     1 0.708
```
So, I do not think that IRR is systematically different depending on whether one of the attributes is tied (while other attributes are randomly assigned fully). 

When you "predict" IRR, however, the results could be noisy depending on your data, as I noted in #38. So, if the predicted IRR does not look good, it may be safer to use 0.74.

**Reply by brueckmann** [reply]
Thank you so much, that's amazing! I am really grateful!
